src/components/recipes/RecipeForm.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import {useNavigate} from "react-router-dom";
+import {CardWrapper, CardHeader, CardBody} from '../../components/base/Card'
+import {Button} from '../../components/base/Button'
+import {InputForm} from '../../components/base/Input'
+import IngredientsList from "../../components/ingredients/IngredientsList";
+import useInput from '../../hooks/useInput'
+import useAxios from '../../hooks/useAxios'
+
+export default function RecipeForm(props) {
+  let navigate = useNavigate();
+
+  const DOMAIN_URL = '/recipes/'
+
+  const [name, nameInput, resetName] = useInput({
+    type: 'text',
+    placeholder: 'Name',
+    initial_value: props.recipe.name
+  });
+
+  const [description, descriptionInput, resetDescription] = useInput({
+    type: 'area',
+    placeholder: 'Description',
+    initial_value: props.recipe.description
+  });
+
+  const [new_ingredient_name, newIngredientNameInput, resetIngredientName] = useInput({
+    type: 'text',
+    placeholder: 'New Ingredient',
+  });
+  
+  const [ingredients, setIngredients] = useState(props.recipe.ingredients);
+
+  const url = ('id' in props.recipe) ? DOMAIN_URL + props.recipe.id + '/' : DOMAIN_URL  
+
+  const { dispatch, response, loading, error } = useAxios({
+    method: props.method,
+    url,
+    data: {
+      name,
+      description,
+      ingredients,
+    }
+  }, performAction)
```

Comment:
I have the feeling you're mixing different levels of abstraction, lovely level calls to endpoints with high level UI and business logic

Reply:
Yes, Fter checking it, I see that a service would help to separate the logic between the UI and the hooks